File: server/src/services/chunking.js
function chunkResume(resume) {
  const chunks = [];
  let index = 0;

  if (resume.skills && resume.skills.length > 0) {
    chunks.push({
      chunkText: `Skills: ${resume.skills.join(', ')}`,
      section: 'skills',
      chunkIndex: index++,
    });
  }

  for (const entry of resume.experience || []) {
    chunks.push({ chunkText: entry, section: 'experience', chunkIndex: index++ });
  }

  for (const entry of resume.projects || []) {
    chunks.push({ chunkText: entry, section: 'project', chunkIndex: index++ });
  }

  if (resume.summary) {
    chunks.push({ chunkText: resume.summary, section: 'summary', chunkIndex: index++ });
  }

  return chunks;
}

module.exports = { chunkResume };

File: python-service/app/chunking.py
from langchain_core.documents import Document

def chunk_resume(resume: dict) -> list[Document]:
    chunks = []
    index = 0

    if resume.get("skills"):
        chunks.append(Document(
            page_content=f"Skills: {', '.join(resume['skills'])}",
            metadata={"section": "skills", "chunkIndex": index, "resumeId": resume["resumeId"], "user": resume["user"]}
        ))
        index += 1

    for entry in resume.get("experience", []):
        chunks.append(Document(page_content=entry,
            metadata={"section": "experience", "chunkIndex": index, "resumeId": resume["resumeId"], "user": resume["user"]}))
        index += 1

    for entry in resume.get("projects", []):
        chunks.append(Document(page_content=entry,
            metadata={"section": "project", "chunkIndex": index, "resumeId": resume["resumeId"], "user": resume["user"]}))
        index += 1

    if resume.get("summary"):
        chunks.append(Document(page_content=resume["summary"],
            metadata={"section": "summary", "chunkIndex": index, "resumeId": resume["resumeId"], "user": resume["user"]}))
        index += 1

    return chunks